compiler/rustc_trait_selection/src/traits/project.rs

```diff
@@ -753,8 +781,7 @@ impl<'tcx> TypeFolder<'tcx> for BoundVarReplacer<'_, 'tcx> {
                     .tcx
                     .mk_const(ty::ConstS { kind: ty::ConstKind::Placeholder(p), ty: ct.ty() })
             }
-            _ if ct.has_vars_bound_at_or_above(self.current_index) => ct.super_fold_with(self),
-            _ => ct,
+            _ => ct.super_fold_with(self),
```

Comment:
Was the code before...wrong? Should be just a perf optimization (don't need to fold if there aren't and bound vars to replace.